src/plugins/part.c

```diff
@@ -779,7 +793,7 @@ static gboolean resize_part (PedPartition *part, PedDevice *dev, PedDisk *disk,
         ped_constraint_destroy (constr);
         finish_alignment_constraint (disk, orig_flag_state);
         return FALSE;
-    } else if (part->geom.start != start || part->geom.length < new_size) {
+    } else if (part->geom.start != start || part->geom.length < new_size - tolerance) {
```

Comment:
Would be great if we had different error messages for wrong start and size. Otherwise it's hard to see what went wrong from the error message.